Inc/HALALMock/Services/SharedMemory/SharedMemory.hpp

```diff
@@ -9,7 +10,7 @@ class SharedMemory {
     constexpr static uint8_t max_pin_size_bytes{16};
 
     // do not use before SharedMemory::start, uninitialized!!!
-    static void *gpio_memory;
+    static PinModel *gpio_memory;
```

Comment:
make this private and create a new function to access it, something like:
```c++
static PinModel &get_pin(Pin pin);
```
this prevents code from messing with the base pointer, since once created you just want to access it

Reply:
The interface I proposed passed a reference, not a value, maybe you didn't catch it because of the sintax?, as your preference it would be:
```c++
static PinModel& get_pin(Pin pin);
```